src/jaegertracing/testutils/MockAgent.cpp

```diff
@@ -31,10 +31,36 @@
 #include "jaegertracing/utils/ErrorUtil.h"
 #include "jaegertracing/utils/UDPClient.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable : 4267)
+#pragma warning(disable : 4244)
+#endif
+
 namespace jaegertracing {
 namespace testutils {
 namespace {
 
+struct SocketUtils {
+    static size_t read(int socketHandle, char* buffer, size_t size) {
+#ifdef WIN32
+        int readBytes = ::recv(socketHandle, buffer, size, 0);
+        return std::max(0, readBytes); 
+#else
+        return ::read(socketHandle, buffer, size);
+#endif  
+    }
+
+    static size_t write(int socketHandle, const char* msg, size_t size)
+    {
+#ifdef WIN32
+        return ::send(socketHandle, msg, size, 0);
```

Comment:
`send` and `recv` work on Linux/Mac too. You can just use the Windows approach for all platforms and delete the original `read`/`write` calls.

Reply:
done